friends/src/actions/index.js

```diff
@@ -0,0 +1,41 @@
+import axios from 'axios';
+
+export const FRIENDS_FETCH_SUCCESS = 'FRIENDS_FETCH_SUCCESS';
+export const FRIENDS_FETCH_FAILURE = 'FRIENDS_FETCH_FAILURE';
+export const FRIENDS_FETCH = 'FRIENDS_FETCH';
+export const FRIENDS_FETCH_FINISHED = 'FRIENDS_FETCH_FINISHED';
+export const ADD_FRIEND = 'ADD_FRIEND';
+export const DELETE_FRIEND = 'DELETE_FRIEND';
+
+export const fetchData = () => dispatch => {
+  dispatch({ type: FRIENDS_FETCH });
+  axios
+    .get('http://localhost:5000/api/friends')
+    .then(response => {
+      dispatch({ type: FRIENDS_FETCH_SUCCESS, payload: response.data });
+      dispatch({ type: FRIENDS_FETCH_FINISHED });
```

Comment:
Nice job decoupling the state into a loadingReducer and friendsReducer. What do you think of the pattern? I also believe it's possible to have loadingReducer listen to same action as friendsReducer to switch the boolean ... like both friendsReducer and loadingReducer will do something on FRIENDS_FETCH_SUCCESS ... the friendsReducer will use the payload to populate the new state of friends ... while the loadingReducer will simple return true and not use the payload but they can share the same action creator as a trigger.